nixos/modules/system/boot/luksroot.nix

```diff
@@ -430,9 +434,22 @@ let
         ${optionalString (lib.versionOlder kernelPackages.kernel.version "5.4") ''
           echo "On systems with Linux Kernel < 5.4, it might take a while to initialize the CRNG, you might want to use linuxPackages_latest."
           echo "Please move your mouse to create needed randomness."
+        ''}
+        ${if dev.fido2.requiresPin then ''
+          mkfifo ${fifo_tmp_fido_pin}
+          read -rsp "FIDO2 pin for ${dev.device}: " fido_pin
+        '' else ''
+          # Dont make a fifo queue or ask for pin since no pin entry is required
         ''}
           echo "Waiting for your FIDO2 device..."
-          fido2luks open ${dev.device} ${dev.name} ${dev.fido2.credential} --await-dev ${toString dev.fido2.gracePeriod} --salt string:$passphrase
+          ${f2lopen} --salt string:$passphrase &
+          f2lopen_pid=$!
+          echo -n $fido_pin > ${fifo_tmp_fido_pin}
+          # Clean up after ourselves
```

Comment:
```suggestion
```

Reply:
Why do you want to remove this comment? Just wondering!

Reply:
It is very obvious that unset and rm clears up after you and obvious comments should be avoided.

Reply:
Thanks, that makes sense.